### ISSUE TITLE
io.socket not persistently reconnect

### ISSUE THREAD

**Issue by meepeek** [balderdashy/sails#2347]
Please help. I would like to know how to configure sails.io.js for these things:
1. Delay between each retry.
2. Making retry persistent.
3. Manually activate io.sails connection if the automation stopped or initialize to the server got rejected (when server is unreachable at the initialize state)

## My Scenario
I am using Sails as a backend server and my client is ionic (cordova). All the communication in between is being done by including sails.io.js to the client side and set io.sails.url = 'http://localhost:1337'; before the initialize of connection.

## The Issue
- If the server is offline or unreachable, ajax error and no more attempt to reconnect and I could not command io.sails to retry. Look at the console, I got this:
```
GET http://localhost:1337/__getcookie net::ERR_CONNECTION_REFUSED sails.io.js:290
```
- If the client reached the server and form a connection, then I made the server unreachable for a while. io.sails made around 10 attempts to reconnect and then gone. I can no longer reactivate the client to has another try, even after the server backed online.

**Comment by cyrilchapon** [comment]
Up.

**Comment by CWyrtzen** [comment]
Is this still an issue?

**Comment by cyrilchapon** [comment]
Yes definately.

> The only problem I still have is that if the server is unreachable at the very beginning, `io.socket.socket` does not even exists so I cannot make a connection attempt.

sails.io.js crashes without any chance to rework without refreshing the page

**Comment by sailsbot** [comment]
Thanks for posting, @meepeek.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
